app/grandchallenge/evaluation/tasks.py

```diff
@@ -562,19 +564,80 @@ def get_average_job_duration_for_phase(phase):
     jobs = Job.objects.filter(
         outputs__evaluation_evaluations_as_input__submission__phase=phase,
     ).distinct()
+    start_date = datetime.datetime.strptime("1/1/2021", "%d/%m/%Y")
+    end_date = datetime.datetime.now()
+    delta = relativedelta.relativedelta(end_date, start_date)
+    monthly_costs = {}
+    algorithms_submitted_per_month = {}
+    for year in [start_date.year, start_date.year + delta.years]:
+        if not year == start_date.year + delta.years:
+            months = range(1, 13)
+        else:
+            months = range(1, 2 + delta.months)
+        for month in months:
+            _, num_days = calendar.monthrange(year, month)
+            job_start_date = datetime.datetime(
+                year, month, 1, 0, 0, 0, tzinfo=datetime.timezone.utc
+            )
+            job_end_date = datetime.datetime(
+                year, month, num_days, 23, 59, 59, tzinfo=datetime.timezone.utc
+            )
+            jobs_for_month = jobs.filter(
+                started_at__gte=job_start_date, started_at__lte=job_end_date
+            )
+            submitted_algorithms = [
+                str(pk)
+                for pk in jobs_for_month.values_list(
+                    "algorithm_image__algorithm__pk", flat=True
+                )
+            ]
+            total_duration = jobs_for_month.total_duration()
+            compute_cost = (
+                round(
+                    total_duration.total_seconds()
+                    * settings.CHALLENGES_COMPUTE_COST_CENTS_PER_HOUR
+                    / 3600
+                    / 100,
+                    ndigits=2,
+                )
+                if total_duration
+                else 0
+            )
+            try:
+                monthly_costs[year][
+                    job_start_date.strftime("%B")
+                ] = compute_cost
+                algorithms_submitted_per_month[year][
+                    job_start_date.strftime("%B")
+                ] = submitted_algorithms
+            except (TypeError, KeyError):
+                monthly_costs[year] = {}
+                algorithms_submitted_per_month[year] = {}
+                monthly_costs[year][
+                    job_start_date.strftime("%B")
+                ] = compute_cost
+                algorithms_submitted_per_month[year][
+                    job_start_date.strftime("%B")
+                ] = submitted_algorithms
+
```

Comment:
This method is now doing a lot more than `get_average_job_duration_for_phase`, probably best to split it out or rename.